cli/src/schema/l2Upgrade.ts

```diff
@@ -1,35 +1,9 @@
 import { z } from "zod";
 import { account20String, bytes32Hash, hashString, transactionSchema } from "../schema";
 
-const systemContractNames = z.enum([
-  "AccountCodeStorage",
-  "NonceHolder",
-  "KnownCodesStorage",
-  "ImmutableSimulator",
-  "ContractDeployer",
-  "L1Messenger",
-  "MsgValueSimulator",
-  "L2EthToken",
-  "Keccak256",
-  "SHA256",
-  "Ecrecover",
-  "EcAdd",
-  "EcMul",
-  "P256Verify",
-  "CodeOracle",
-  "SystemContext",
-  "EventWriter",
-  "BootloaderUtilities",
-  "Compressor",
-  "ComplexUpgrader",
-  "EmptyContract",
-  "PubdataChunkPublisher",
-  "GasBoundCaller",
-]);
-
 const systemContracts = z.array(
   z.object({
-    name: systemContractNames,
+    name: z.string(),
```

Comment:
This is the main fix. Instead of allowing a list of possible contract names, I'm allowing just a string. Contract names can be added, so there is no point in restrict this during the parsing.